### PR TITLE
[v9] Ensure audit logging of tsh app login.

### PR DESCRIPTION
**Pull request by mdwn**
Backport of https://github.com/gravitational/teleport/pull/16827.

### CODE REVIEW

**Review by probakowski — APPROVED** [review]

File: lib/auth/sessions.go [hunk 1]
```diff
@@ -102,6 +108,49 @@ func (s *Server) CreateAppSession(ctx context.Context, req types.CreateAppSessio
...
+
+	// User metadata from the audit event
+	userMetadata := identity.GetUserMetadata()
+	userMetadata.User = session.GetUser()
+	userMetadata.AWSRoleARN = req.AWSRoleARN
+
+	// Record peer for the address of the requesting connection.
+	p, ok := peer.FromContext(ctx)
+	if !ok {
+		return nil, trace.Wrap(errors.New("unable to get peer from context"))
```
Comment (r0mant) [inline lib/auth/sessions.go:120]:
```suggestion
		return nil, trace.BadParameter("unable to get peer from context")
```

File: lib/auth/sessions.go [hunk 2]
```diff
@@ -102,6 +108,49 @@ func (s *Server) CreateAppSession(ctx context.Context, req types.CreateAppSessio
...
 		return nil, trace.Wrap(err)
 	}
+
+	// User metadata from the audit event
+	userMetadata := identity.GetUserMetadata()
+	userMetadata.User = session.GetUser()
+	userMetadata.AWSRoleARN = req.AWSRoleARN
+
+	// Record peer for the address of the requesting connection.
+	p, ok := peer.FromContext(ctx)
```
Comment (r0mant) [inline lib/auth/sessions.go:118]: This is going to be address of proxy making request to this auth server, not the client, right? So it's not really equivalent to what we were logging as the remote addr before.

Comment (mdwn) [reply lib/auth/sessions.go:118]: Running this on `branch/v9` (and other branches) without this patch, we've got:

```
  "addr.remote": "192.168.1.232:51369",
```

With this patch, we've got

```
  "addr.remote": "127.0.0.1:51419",
```

So it looks like you're right -- I was taking the explanation from https://github.com/gravitational/teleport/issues/9536#issuecomment-1219430266 at face value here.

So given that this isn't quite right, what's the process here? Should I revert the change in the master branch, or backport all of these and then backport the fix?

Comment (r0mant) [reply lib/auth/sessions.go:118]: Let's just submit the fix separately to master (no need to reverting anything) and include it with these backports too.

**Review by ravicious — APPROVED** [review]

**Comment by mdwn** [comment]
Closing this for now, as this has a conflict that will be solved by https://github.com/gravitational/teleport-private/pull/167. We need to wait for that to be implemented.

**Pull request closed without merging**